### PR TITLE
Add inline example for grdtrack

### PR DESCRIPTION
**Pull request by willschlitzer**
This PR adds an inline code example to the docstring for `grdtrack`.

Addresses #1686

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/grdtrack.py [hunk 1]
```diff
@@ -264,6 +264,22 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
...
+    >>> # -107, and a y-range of -49 to -42
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[-118, -107, -49, -42]
+    ... )  # doctest: +SKIP
+    >>> # Load a pandas dataframe with ocean ridge points
+    >>> points = pygmt.datasets.load_ocean_ridge_points()  # doctest: +SKIP
+    >>> # Create a pandas dataframe from an input grid and set of points
+    >>> # The output dataframe adds a column named "bathymetry"
+    >>> output_dataframe = pygmt.grdtrack(
+    ...     points=dataframe, grid=grid, newcolname="bathymetry"
```
Comment (michaelgrund) [inline pygmt/src/grdtrack.py:281]:
```suggestion
    ...     points=points, grid=grid, newcolname="bathymetry"
```

**Review by seisman — APPROVED** [review]

File: pygmt/src/grdtrack.py [hunk 2]
```diff
@@ -264,6 +264,22 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
...
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[-118, -107, -49, -42]
+    ... )  # doctest: +SKIP
+    >>> # Load a pandas dataframe with ocean ridge points
+    >>> points = pygmt.datasets.load_ocean_ridge_points()  # doctest: +SKIP
+    >>> # Create a pandas dataframe from an input grid and set of points
+    >>> # The output dataframe adds a column named "bathymetry"
+    >>> output_dataframe = pygmt.grdtrack(
+    ...     points=points, grid=grid, newcolname="bathymetry"
+    ... )  # doctest: +SKIP
```
Comment (seisman) [inline pygmt/src/grdtrack.py:282]:
```suggestion
    ... )
```

**Comment by willschlitzer** [comment]
@seisman I'm guessing I'm missing something, but why are you suggesting removing all of the `# doctest: +SKIP` from the inline example?

**Comment by seisman** [comment]
> @seisman I'm guessing I'm missing something, but why are you suggesting removing all of the `# doctest: +SKIP` from the inline example?

Yes, we have adopted a new doctest method in https://github.com/GenericMappingTools/pygmt/pull/1790.

**Review by seisman — APPROVED** [review]

File: pygmt/src/grdtrack.py [hunk 3]
```diff
@@ -264,6 +266,22 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
...
+    Example
+    -------
+    >>> import pygmt
+    >>> # Load a grid of @earth_relief_30m data, with an x-range of -118 to
+    >>> # -107, and a y-range of -49 to -42
+    >>> grid = pygmt.datasets.load_earth_relief(
+    ...     resolution="30m", region=[-118, -107, -49, -42]
+    ... )
+    >>> # Load a pandas dataframe with ocean ridge points
+    >>> points = pygmt.datasets.load_sample_data(name='ocean_ridge_points')
```
Comment (meghanrjones) [inline pygmt/src/grdtrack.py:279]:
```suggestion
    >>> points = pygmt.datasets.load_sample_data(name="ocean_ridge_points")
```
To fix the style checks workflow.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**